packages/cardpay-sdk/sdk/utils/safe-utils.ts

```diff
@@ -477,18 +477,55 @@ export function gasInToken(estimate: Estimate): BN {
 // https://github.com/cardstack/card-protocol-relay-service/blob/master/safe.py#L303-L316
 export const baseGasBuffer: BN = new BN('30');
 
+interface BalanceEntry {
+  tokenAddress: string | null;
+  balance: string;
+  token: {
+    decimals: number;
+    logoUri: string;
+    name: string;
+    symbol: string;
+  } | null;
+}
+
+interface BalanceSummary {
+  tokenAddress: string;
+  symbol: string;
+  balance: BigNumber;
+  decimals: number;
+}
+
 export async function getTokenBalancesForSafe(
   provider: JsonRpcProvider,
   tokenAddresses: string[],
   safeAddress: string
-) {
-  return await Promise.all(
-    tokenAddresses.map(async (tokenAddress: string) => {
-      let token = new Contract(tokenAddress, ERC20ABI, provider);
-      let symbol = await token.symbol();
-      let balance = await token.callStatic.balanceOf(safeAddress);
-      let decimals = await token.decimals();
-      return { tokenAddress, symbol, balance, decimals };
+): Promise<BalanceSummary[]> {
+  let baseUrl = await getConstant('safeTransactionServiceUrl', provider);
+
+  if (!baseUrl) {
+    throw new Error('Could not find safe transaction service url for current network');
+  }
+  let response = await fetch(`${baseUrl}/api/v1/safes/${safeAddress}/balances/`);
+
+  if (!response?.ok) {
+    throw new Error(await response.text());
+  }
+  let balances: BalanceEntry[] = await response.json();
+
+  return tokenAddresses
+    .map((tokenAddress) => {
+      let balanceInfo = balances.find((b) => b.tokenAddress?.toLowerCase() === tokenAddress.toLowerCase());
+
+      if (balanceInfo?.token) {
+        let {
+          balance,
+          token: { symbol, decimals },
+        } = balanceInfo;
+
+        return { tokenAddress, symbol, balance: BigNumber.from(balance), decimals };
+      } else {
+        return null;
+      }
     })
-  );
+    .filter((e) => !!e) as BalanceSummary[];
```

Comment:
if you use a function like this `const nonNullable = <T>(value: T): value is NonNullable<T> => !!value;`  typescript will know the value is valid and you won't need to cast anything.
then 
`.filter(nonNullable)`

it would be nice to have this as a util in the sdk, since we also do the same stuff on the wallet

Reply:
That's so much better, I always feel guilty using `as` :)